core/plugins/org.polarsys.capella.core.projection.scenario/src/org/polarsys/capella/core/projection/scenario/topdown/TopDownExt.java

```diff
@@ -0,0 +1,399 @@
+      if (instance != null) {
+        partBounds.add(instance);
+      } else {
+        UnwantedObjects.add(end.getEvent(), transfo_p);
+      }
+    }
+
+    if ((partBounds.size() == 0) && (role.getRepresentedInstance() != null)) {
+      String messageTxxt = Messages.Scenario_MultipleTransitionedElements;
+      List<? extends EObject> availables = Query.retrieveTransformedElements(role.getRepresentedInstance(), transfo_p,
+          InformationPackage.Literals.ABSTRACT_INSTANCE);
+      if ((availables.size() == 0) && (role.getRepresentedInstance() instanceof Part)) {
+        for (Component element : (List<Component>) Query.retrieveTransformedElements(
+            role.getRepresentedInstance().getAbstractType(), transfo_p, CsPackage.Literals.COMPONENT)) {
+          availables.addAll((List) element.getRepresentingParts());
+        }
+      }
+
+      if (availables.size() <= 1) {
+        partBounds.addAll((List) availables);
+      } else {
+        IResolver resolver = ResolverFinalizer.getResolver(transfo_p);
+        if (resolver != null) {
+          List<EObject> res = resolver.resolve(role.getRepresentedInstance(), (List) availables,
+              ScenarioTransfo.getTitle(transfo_p), messageTxxt, false, transfo_p,
+              new EObject[] { role.getRepresentedInstance() });
+          if ((res != null) && (res.size() > 0)) {
+            partBounds.add((AbstractInstance) res.get(0));
+          }
+        }
+      }
+    }
+
+    partBounds = (List) ListExt.removeDuplicates((List) partBounds);
+
+    return partBounds;
+  }
+
+  /**
+   * Retrieve an instance for the given end and given operation_p. sourceEnd is used to restrict
+   */
+  @SuppressWarnings({ "unchecked", "rawtypes" })
+  public static AbstractInstance getRelatedInstance(boolean isSource, AbstractEnd sourceEnd,
+      AbstractEventOperation targetOperation, ITransfo transfo_p) {
+    List<AbstractInstance> partBounds = new ArrayList<AbstractInstance>();
+
+    if (sourceEnd != null) {
+
+      if (!ScenarioExt.isExchangeItemInstanceRole(sourceEnd)) {
+        Collection<AbstractInstance> relatedParts = getRelatedInstances(isSource, targetOperation, transfo_p);
+
+        // Restrict to components or parent components.
+        boolean isLinked = false;
+        for (AbstractInstance relatedPart : relatedParts) {
+          for (AbstractInstance parent : getRelatedPartAndAncestors(relatedPart)) {
+            for (InstanceRole relatedRole : sourceEnd.getCoveredInstanceRoles()) {
+              if (RefinementLinkExt.isLinkedTo(parent, relatedRole.getRepresentedInstance())) {
+                isLinked = true;
+                partBounds.add(relatedPart);
+              }
+            }
+          }
+        }
+
+        if (!isLinked) {
+          partBounds.addAll(relatedParts);
+        }
+
+      } else {
+        ExchangeItemInstance instance = (ExchangeItemInstance) ScenarioExt.getExchangeItemInstanceRole(sourceEnd);
```

Comment:
To make ExchangeItem as instance role work, its here. 
targetOperation is the ExchangeItemAllocation to use (the one from targetArchitecture if it exist, or the original one if not transitioned according to preferences)
We lookup the original ExchangeItemInstance of the message end. If targetOperation is transitioned, then we lookup for the transitioned ExchangeItemInstance. Otherwise, we use the ExchangeItemInstance if it match with targetOperation